container/image.bzl

```diff
@@ -770,7 +770,15 @@ _outputs["config_digest"] = "%{name}.json.sha256"
 _outputs["build_script"] = "%{name}.executable"
 
 def _image_transition_impl(settings, attr):
-    return dicts.add(settings, {
+    if not settings["@io_bazel_rules_docker//transitions:enable"]:
+        # Once bazel < 5.0 is not supported we can return an empty dict here
+        return {
+            "//command_line_option:platforms": settings["//command_line_option:platforms"],
+            "@io_bazel_rules_docker//platforms:image_transition_cpu": "//plaftorms:image_transition_cpu_unset",
+            "@io_bazel_rules_docker//platforms:image_transition_os": "//plaftorms:image_transition_os_unset",
```

Comment:
Hey there  @laurynaslubys, First of all thanks a bunch for making these changes and my team has been really interested in something to address some of the new overhead caused by transitions. I wanted to mention 2 things:

I noticed this when I was trying to debug my own issue. But it looks like there is a typo here `plaftorm` instead of `platform`.

I think one of the main reasons my team is interested in this MR is because it could reduce the number of rebuilds. Since 0.23, we will basically have to rebuild/recompile our entire pipeline, even though our host and target are the same, because I think the transition settings get hashed into the built directory. (i.e. our bazel-out directory `k8-fastbuild` becomes `k8-fastbuild-ST-<someHashHere.`). I'm happy to open an issue, but I would really like if things could behave as before in rules_docker 0.22 where if you didn't explicitly specify something, it didn't create a new output directory for docker images.

I noticed that having it the `transitions:enable` flag be false causing this function to return an empty JSON object works. But it sounds like maybe that's an issue for people below Bazel 5.0. 

This is still a great change and my no means want to downplay it.

Reply:
@cfife-btig Thank you for pointing this out. I spent several hours tracking down the exact same cache miss issue. Patching this to return an empty dict on 0.25.0 and turning transitions off has resolved my issue for now.

Reply:
Hi @scasagrande !  I just tracked it down today, only our version of Bazel doesn't seem to be compatible with the empty dict solution, so I'm looking at 0.22.

Reply:
Just a dumb sanity check @abentley-ssimwave but are you setting `io_bazel_rules_docker/transitions:enable=false` on the commandline?

Reply:
@abentley-ssimwave well hello!! Yes I should have included my bazel version, we were on 5.2 at time of my prior comment and we're now up to 5.3.1

Reply:
@cfife-btig Without your patch, `io_bazel_rules_docker/transitions:enable=false` changed the directory but didn't make it match other targets.  One of my teammates is trying to migrate to the latest Bazel but for the moment we're on 3.4.1.  Also we're using Dazel, a wrapper to run under Docker.  Anyhow, I get this:
```
$ dazel build <super-secret> --@io_bazel_rules_docker//transitions:enable=false
ERROR: /home/abentley/.dazel-cache/_dazel_buildnode_20_abentley/0f41bef669d02a975c1e566c67e15b8d/external/io_bazel_rules_docker/container/image.bzl:769:5: transition outputs [//command_line_option:platforms, @io_bazel_rules_docker//platforms:image_transition_cpu, @io_bazel_rules_docker//platforms:image_transition_os] were not defined by transition function
ERROR: Errors encountered while applying Starlark transition
ERROR: /home/abentley/.dazel-cache/_dazel_buildnode_20_abentley/0f41bef669d02a975c1e566c67e15b8d/external/io_bazel_rules_docker/container/image.bzl:769:5: transition outputs [//command_line_option:platforms, @io_bazel_rules_docker//platforms:image_transition_cpu, @io_bazel_rules_docker//platforms:image_transition_os] were not defined by transition function
ERROR: Errors encountered while applying Starlark transition
ERROR: /home/abentley/.dazel-cache/_dazel_buildnode_20_abentley/0f41bef669d02a975c1e566c67e15b8d/external/io_bazel_rules_docker/container/image.bzl:769:5: transition outputs [//command_line_option:platforms, @io_bazel_rules_docker//platforms:image_transition_cpu, @io_bazel_rules_docker//platforms:image_transition_os] were not defined by transition function
ERROR: Errors encountered while applying Starlark transition
ERROR: Analysis of target '<super-secret> failed; build aborted: com.google.devtools.build.lib.analysis.starlark.StarlarkTransition$TransitionException: Errors encountered while applying Starlark transition
```

Reply:
That sounds like what should happen. You need to both use the patch and set the flag transitions:enable=false. Sorry if this is redundant, but if you're not doing both of those things it won't disable transitions.

If you only set the flag, it just sets the directory `<myoutdir>-hash` to `<myoutdir>-somenewhash` because removing those build settings are just generating a new transition hash (when it should really remove the transition has altogether), whereas in 0.22, I believe there were no transitions to begin with.

Reply:
@cfife-btig Yes, I set `--@io_bazel_rules_docker//transitions:enable=false` you can see it in the output I posted.

Reply:
@cfife-btig The result of using the flag with your patch on 3.4.1 is a whole pile of errors, and I do not agree that that "should" happen, although I can accept that there may not be a better solution for 0.25

Reply:
Hmmm. I'm not sure then. My guess would be your version of Bazel is the issue. I think I'm on 5.2

Basically the patch file returns an empty dict instead of `unset`, and i didn't write it but there's a comment implying it might not be supported in < 5.0 Bazel.

```
         # Once bazel < 5.0 is not supported we can return an empty dict here
-        return {
-            "//command_line_option:platforms": settings["//command_line_option:platforms"],
-            "@io_bazel_rules_docker//platforms:image_transition_cpu": "//plaftorms:image_transition_cpu_unset",
-            "@io_bazel_rules_docker//platforms:image_transition_os": "//plaftorms:image_transition_os_unset",
-        }
+        return {}
```

Reply:
@cfife-btig Yes, I agree.